lib/mauth/rack.rb

```diff
@@ -70,11 +86,20 @@ def should_authenticate?(env)
       end
     end
 
-    # signs outgoing responses
+    # signs outgoing responses with only the protocol used to sign the request.
     class ResponseSigner < MAuth::Middleware
       def call(env)
+        # QUESTION does instaniate a new request object, rather than doing a check for the headers here, add too much overhead?
+        versioned_token = MAuth::Rack::Request.new(env).signature_token
         unsigned_response = @app.call(env)
-        signed_response = mauth_client.signed(MAuth::Rack::Response.new(*unsigned_response))
+
+        signed_response =
+          if versioned_token == MAuth::Client::MWSV2_TOKEN
+            mauth_client.signed(MAuth::Rack::Response.new(*unsigned_response), v2_only_override: true)
+          else
+            mauth_client.signed(MAuth::Rack::Response.new(*unsigned_response), v1_only_override: true)
```

Comment:
Not a fan of these override flags.  I'd rather see just two different function on the client like `signed_v1` and `signed_v2`.

Reply:
With that and the above about sticking the version in env, think this could be something cleaner like:
```ruby
method = env['mauth.request_version'] == 2 ? :signed_v2 : :signed_v1
response = mauth_client.send(method, MAuth::Rack::Response.new(*unsigned_response))
response.status_headers_body
```